Assets/MixedRealityToolkit.Services/InputSystem/BaseNearInteractionTouchable.cs

```diff
@@ -29,57 +26,47 @@ public abstract class BaseNearInteractionTouchable : MonoBehaviour
         /// </summary>
         public TouchableEventType EventsToReceive => eventsToReceive;
 
-        public bool ColliderEnabled { get { return !usesCollider || touchableCollider.enabled && touchableCollider.gameObject.activeInHierarchy; } }
-
-        /// <summary>
-        /// False if no collider is found on validate.
-        /// This is used to avoid the perf cost of a null check with the collider.
-        /// </summary>
-        protected bool usesCollider = false;
-
-        /// <summary>
-        /// The collider used by this touchable.
-        /// </summary>
-        [SerializeField]
-        [FormerlySerializedAs("collider")]
-        protected Collider touchableCollider;
-
-        protected float distFront = 0.2f;
-
-        [Tooltip("Distance behind the surface at which you will receive a touch up event")]
+        [Tooltip("Distance behind the surface at which you will receive a touch started event")]
         [SerializeField]
-        protected float distBack = 0.25f;
+        [FormerlySerializedAs("distBack")]
+        protected float pokeThreshold = 0.25f;
+        public float PokeThreshold => pokeThreshold;
 
-        [Tooltip("Distance in front of the surface at which you will receive a touch up event")]
+        [Tooltip("Distance in front of the surface at which you will receive a touch completed event")]
         [SerializeField]
         protected float debounceThreshold = 0.01f;
-
-        public float DistBack => distBack;
-
-        public float DistFront => distFront;
-
         public float DebounceThreshold => debounceThreshold;
 
-        protected void OnEnable()
+        protected void OnValidate()
         {
-            instances.Add(this);
+            pokeThreshold = Math.Max(pokeThreshold, 0);
+            debounceThreshold = Math.Max(debounceThreshold, 0);
         }
 
-        protected void OnDisable()
-        {
-            instances.Remove(this);
-        }
+        public abstract float DistanceToTouchable(Vector3 samplePoint, out Vector3 normal);
+    }
 
-        protected void OnValidate()
+    /// <summary>
+    /// Base class for all touchables using colliders
+    /// </summary>
+    [RequireComponent(typeof(Collider))]
+    public abstract class ColliderNearInteractionTouchable : BaseNearInteractionTouchable
```

Comment:
It seems that with these changes all near interaction touchables must have a collider. Does it make sense to keep a separate ColliderNearInteractionTouchable class?

Reply:
See comment below about NearInteractionTouchableUnityUI. I would love to kick out the UnityUI class, at which point we could indeed merge the base classes too.

Reply:
Never mind, UnityUI case seems to be too important to drop.